src/modules/bigquery.js

```diff
@@ -6,101 +6,56 @@ export default class BigQuery {
     this.pageTokenMap = {};
   }
 
-  pageData = (query, queryResults, columns, token) =>
-    new Promise(resolve => {
-      let rawData = {};
-      const data = [];
-
-      const { jobId, projectId } = queryResults.jobReference;
-
-      if (query.tokenToUse === undefined && query.page === 0) {
-        this.pageTokenMap[1] = queryResults.pageToken;
-      }
-
-      if (this.pageTokenMap[query.page] === undefined && query.page !== 0) {
-        this.pageTokenMap[query.page] = query.tokenToUse;
-      }
-
-      const url = `https://bigquery.googleapis.com/bigquery/v2/projects/${projectId}/queries/${jobId}`;
-      const params = {
-        maxResults: query.pageSize,
-        pageToken: this.pageTokenMap[query.page],
-      };
-
-      axios
-        .get(url, {
-          headers: {
-            'Content-Type': 'application/json',
-            Authorization: `Bearer ${token}`,
-          },
-          params,
-        })
-        .then(response => {
-          rawData = response.data;
-          query.tokenToUse = rawData.pageToken;
+  transformColumns = queryResults =>
+    _.get(queryResults, 'schema.fields', []).map(field => {
+      return { id: field.name, label: field.name, minWidth: 100, type: field.type };
+    });
 
-          if (rawData.rows && rawData.rows.length > 0) {
-            rawData.rows.forEach(rowData => {
-              const row = {};
+  transformRows = (queryResults, columns) => {
+    let rows = _.get(queryResults, 'rows', []).map(row => {
+      return _.get(row, 'f', []).map(value => {
+        return _.get(value, 'v', '');
+      });
+    });
 
-              for (let i = 0; i < rowData.f.length; i++) {
-                const currColumn = columns[i];
-                let item = rowData.f[i].v;
+    return rows.map(row => this.createData(columns, row));
+  };
 
-                if (currColumn.datatype === 'integer') {
-                  item = this.commaFormatted(item);
-                }
+  createData = (columns, row) => {
+    let i = 0;
+    const res = {};
+    for (i = 0; i < columns.length; i++) {
+      const column = columns[i];
+      const columnId = _.get(column, 'id');
```

Comment:
What does this gain us over using `column.id`? I can see usefulness in the cases where the properties are nested inside of the object, but this seems unnecessary.